src/screens/WaitTimeScreen.tsx

```diff
@@ -1,42 +1,471 @@
+              </View>
+              <View>
+                <Button
+                  style={styles.sortButtons}
+                  onPress={() => {
+                    setIsVisible(false);
+                    setSortBy("Time");
+                  }}
+                >
+                  <StyledText>Time</StyledText>
+                  <View style={styles.checkIconTime}>
+                    {sortBy === "Time" && (
+                      <Feather name="check" size={24} color="black" />
+                    )}
+                  </View>
+                </Button>
+                <Button
+                  style={styles.sortButtons}
+                  onPress={() => {
+                    setIsVisible(false);
+                    setSortBy("Distance");
+                  }}
+                >
+                  <StyledText>Distance</StyledText>
+                  <View style={styles.checkIconDistance}>
+                    {sortBy === "Distance" && (
+                      <Feather name="check" size={24} color="black" />
+                    )}
+                  </View>
+                </Button>
+              </View>
+            </View>
+          </TouchableWithoutFeedback>
+        </View>
+      </TouchableWithoutFeedback>
+    </Modal>
+  );
+
+  let searchBar: SearchBar | null;
   return (
-    <View style={styles.container}>
-      <StyledText style={styles.huge}>Wait Times</StyledText>
-      <WhiteSpace />
-      <Button
-        type={"primary"}
-        onPress={() =>
-          navigation.navigate(LOCATION_DETAILS, { location: "Location X" })
-        }
-      >
-        <StyledText>Go to Location X</StyledText>
-      </Button>
-      <StatusBar style="auto" />
-    </View>
+    <TouchableWithoutFeedback
+      onPress={() => {
+        searchBar?.inputRef?.blur();
+      }}
+    >
+      <View style={theme.styles.screenContainer}>
+        <View style={styles.searchContainer}>
+          <SearchBar
+            ref={(el) => ((searchBar as any) = el)}
+            placeholderTextColor={theme.styles.placeholderText.color}
+            style={styles.searchBar}
+            value={searchValue}
+            placeholder="Search places"
+            onChange={(value) => {
+              setSearchValue(value);
+            }}
+            onSubmit={(value) => {
+              setSearchValue(value);
+            }}
+            cancelText={"Cancel"}
+            onCancel={() => {
+              searchBar?.inputRef?.blur();
+            }}
+          />
+        </View>
+        <View style={styles.sortContainer}>
+          <ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+            <Button
+              style={styles.sortByTag}
+              onPress={() => {
+                setIsVisible(true);
+              }}
+            >
+              <StyledText style={styles.sortByText}>Sort By</StyledText>
+            </Button>
+            {poiTypes.map((type: string, index: number) => {
+              return (
```

Comment:
nit: You don't need the return here.

Reply:
Apparently, you do, it gives a warning about the child component for scroll view